docs-ref-services/latest/msal-python-overview.md

```diff
@@ -0,0 +1,156 @@
+)
+
+
+1. MSAL proposes a clean separation between
+   [public client applications, and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1).
+   So you will first create either a `PublicClientApplication` or a `ConfidentialClientApplication` instance,
+   and ideally reuse it during the lifecycle of your app. The following example shows a `PublicClientApplication`:
+
+   ```python
+   from msal import PublicClientApplication
+   app = PublicClientApplication(
+       "your_client_id",
+       authority="https://login.microsoftonline.com/Enter_the_Tenant_Name_Here")
+   ```
+
+   Later, each time you would want an access token, you start by:
+   ```python
+   result = None  # It is just an initial value. Please follow instructions below.
+   ```
+
+2. The API model in MSAL provides you explicit control on how to utilize token cache.
+   This cache part is technically optional, but we highly recommend you to harness the power of MSAL cache.
+   It will automatically handle the token refresh for you.
+
+   ```python
+   # We now check the cache to see
+   # whether we already have some accounts that the end user already used to sign in before.
+   accounts = app.get_accounts()
+   if accounts:
+       # If so, you could then somehow display these accounts and let end user choose
+       print("Pick the account you want to use to proceed:")
+       for a in accounts:
+           print(a["username"])
+       # Assuming the end user chose this one
+       chosen = accounts[0]
+       # Now let's try to find a token in cache for this account
+       result = app.acquire_token_silent(["your_scope"], account=chosen)
+   ```
+
+3. Either there is no suitable token in the cache, or you chose to skip the previous step,
+   now it is time to actually send a request to AAD to obtain a token.
+   There are different methods based on your client type and scenario. Here we demonstrate a placeholder flow.
+
+   ```python
+   if not result:
+       # So no suitable token exists in cache. Let's get a new one from AAD.
+       result = app.acquire_token_by_one_of_the_actual_method(..., scopes=["User.Read"])
+   if "access_token" in result:
+       print(result["access_token"])  # Yay!
+   else:
+       print(result.get("error"))
+       print(result.get("error_description"))
+       print(result.get("correlation_id"))  # You may need this when reporting a bug
+   ```
+
+Refer the [Wiki](https://github.com/AzureAD/microsoft-authentication-library-for-python/wiki) pages for more details on the MSAL Python functionality and usage.
+
+## Migrating from ADAL
+
+If your application is using ADAL Python, we recommend you to update to use MSAL Python. No new feature work will be done in ADAL Python.
+
+See the [ADAL to MSAL migration](https://github.com/AzureAD/microsoft-authentication-library-for-python/wiki/Migrate-to-MSAL-Python) guide.
```

Comment:
Haven't looked at it, but this migration doc looks like an article that should be on docs.

Cc: @rayluo

Reply:
We already have that as [an article on docs](https://docs.microsoft.com/en-us/azure/active-directory/develop/migrate-python-adal-msal), although the two seem not currently in sync. I can go ahead to:
1. Go through the existing wiki page, and move some contents into the docs article by a new PR. (Feel free to create a workitem referencing this comment, and assign it to me.)
2. And then I'll retire the existing wiki page, by changing the wiki sidebar to point to the docs article.

Reply:
ADO work item created : https://identitydivision.visualstudio.com/Technical%20Content/_workitems/edit/1290331/

Reply:
Switch the link to point to https://docs.microsoft.com/en-us/azure/active-directory/develop/migrate-python-adal-msal